scripts/inline_comments_data_collection.py

```diff
@@ -0,0 +1,172 @@
+
+    file_path = f"comments/{patch_id}.json"
+    if os.path.exists(file_path) or not resolved_comments:
+        return
+
+    with open(file_path, "w") as f:
+        json.dump([comment.__dict__ for comment in resolved_comments], f, indent=4)
+
+
+def find_recent_update(transactions, comment_date_modified):
+    updates = [
+        transaction
+        for transaction in transactions
+        if transaction["type"] == "update"
+        and transaction["dateModified"] <= comment_date_modified
+    ]
+    if not updates:
+        return None
+    most_recent_update = max(
+        updates, key=lambda transaction: transaction["dateModified"]
+    )
+    return most_recent_update
+
+
+def save_to_dataset(data):
+    dataset_file = "dataset/inline_comment_dataset.json"
+    with open(dataset_file, "a") as f:
+        f.write(json.dumps(data) + "\n")
+
+
+def to_int(value):
+    if not value:
+        return None
+    if not isinstance(value, int):
+        return int(value)
+    return value
+
+
+def process_comments(patch_threshold, diff_length_threshold):
+    comments_dir = "comments"
+    patch_count = 0
+    for file_name in os.listdir(comments_dir):
+        file_path = os.path.join(comments_dir, file_name)
+        with open(file_path, "r") as f:
+            patch_id = int(file_name.replace(".json", ""))
+            transactions = find_transactions_from_patch(patch_id)
+
+            comments = json.load(f)
+            for comment in comments:
+                comment_date_modified = comment["date_modified"]
+                most_recent_update = find_recent_update(
+                    transactions, comment_date_modified
+                )
+                if not most_recent_update:
+                    continue
+
+                fix_patch_id, revision_phid = get_diff_info_from_phid(
+                    most_recent_update["fields"]["new"]
+                )
+
+                # If the most recent patch is the original patch itself, skip it
+                if fix_patch_id == patch_id:
+                    continue
+
+                bug_id = find_bugid_from_revision_phid(phid=revision_phid)
+                review_data.load_patch_by_id(fix_patch_id)
+
+                with open(f"patches/{fix_patch_id}.patch", "r") as f:
+                    patch_diff = f.read()
+
+                if len(patch_diff) > diff_length_threshold:
+                    continue
+
+                data = {
+                    "bug_id": to_int(bug_id),
+                    "revision_phid": revision_phid,
+                    "initial_patch_id": to_int(patch_id),
+                    "fix_patch_id": to_int(fix_patch_id),
+                    "comment": comment,
+                    "fix_patch_diff": patch_diff,
+                }
+                save_to_dataset(data)
+
+        patch_count += 1
+        if patch_count >= patch_threshold:
+            break
+
+
+if __name__ == "__main__":
+    download_inline_comments()
```

Comment:
It is not ideal to save all comments on the desk before processing. Why do we need to do this?

Reply:
Originally I had downloaded all comments to reduce the time to run the script multiple times, however for general use it makes sense to avoid that. Removed it here: 3fffb4900f54ffa5799a7b465e73d027b6496d6e

Reply:
That is a valid use case. It could be an option to have a flag to enable caching of requests when we debug locally.